anyway/views/news_flash/api.py

```diff
@@ -9,13 +9,39 @@
 from anyway.backend_constants import BE_CONST
 from anyway.base import user_optional
 from anyway.models import NewsFlash
+from anyway.infographics_utils import is_news_flash_resolution_supported
 
 DEFAULT_OFFSET_REQ_PARAMETER = 0
 DEFAULT_LIMIT_REQ_PARAMETER = 100
 
+
 @user_optional
 def news_flash():
-    logging.debug("getting news flash")
+    news_flash_id = request.values.get("id")
+
+    if news_flash_id is not None:
+        query = db.session.query(NewsFlash)
+        news_flash_obj = query.filter(NewsFlash.id == news_flash_id).first()
+        if news_flash_obj is not None:
+            if is_news_flash_resolution_supported(news_flash_obj):
+                return Response(
+                    json.dumps(query.serialize(), default=str),
+                    mimetype="application/json"
+                )
+            else:
+                return Response("News flash location not supported", 406)
+        return Response(status=404)
+
+    query = gen_news_flash_query(db.session)
+    news_flashes = query.all()
+
+    news_flashes_jsons = [n.serialize() for n in news_flashes]
+    for news_flash in news_flashes_jsons:
+        set_display_source(news_flash, news_flash_id)
+    return Response(json.dumps(news_flashes_jsons, default=str), mimetype="application/json")
+
+
+def gen_news_flash_query(session):
```

Comment:
I am not sure how i missed it - Sometimes the function `gen_news_flash_query` return an object of type `Response` like in line 60, and sometimes an a SQL query object - like in line 106.

Is this indented?
Maybe add type hinting to the function

Reply:
Good catch!